src/resources.cpp

```diff
@@ -118,15 +122,52 @@ P<ResourceStream> DirectoryResourceProvider::getResourceStream(string filename)
 std::vector<string> DirectoryResourceProvider::findResources(string searchPattern)
 {
     std::vector<string> found_files;
+#if defined(ANDROID)
+    //Limitation : 
+    //As far as I know, Android NDK won't provide a way to list subdirectories
```

Comment:
If I remember right, the AAssetManager_* functions always return all subdirectory files as well?

Reply:
"AAssetDir provides access to a chunk of the asset hierarchy as if it were a single directory. " in NDK doc would pretend you're right... But some posts in stackoverflow says the opposite https://stackoverflow.com/questions/46225251/file-hierarchy-with-android-native-aassetmanager

Also, the code seems not recursive  https://android.googlesource.com/platform/frameworks/base.git/+/1cadc07dd1e3711fb1e57548038e3026682c5ef8/native/android/asset_manager.cpp#135
I also tried dumping the filenames... And only got the first level.
This would also cause some issues in case of same filename in different directories as assets are in the apk in their directories when dumping the apk, and not renamed.

I'm pretty sure it won't return any subdirectory files.
In order to return directories, if I believe stackoverflow, the best is to use zlib 
https://stackoverflow.com/questions/14563943/use-zlib-to-list-directory-structure-of-apk-file-in-android

As a sidenote, checking dir pointer is useless as it will always be valuated, but it cannot harm to let the test...

Reply:
Could fix this after though if needed, as I planned to use this to have language selection on option screen...